usecase/user.go

```diff
@@ -0,0 +1,41 @@
+package usecase
+
+import (
+	"context"
+
+	"github.com/OLIENTTECH/backend-challenges/infrastructure/datastore"
+	rdb "github.com/OLIENTTECH/backend-challenges/infrastructure/external/db/postgres"
+	"github.com/OLIENTTECH/backend-challenges/internal/cerror"
+	"github.com/OLIENTTECH/backend-challenges/pkg/log"
+	"github.com/OLIENTTECH/backend-challenges/usecase/output"
+)
+
+type Users interface {
+	List(ctx context.Context) (*output.ListUsers, error)
+}
+
+func NewUsersUsecase(tx rdb.TxManager, ds datastore.DataStore, logger *log.Logger) Users {
+	return &userUsecase{
+		tx:     tx,
+		ds:     ds,
+		logger: logger,
+	}
+}
+
+func (u *userUsecase) UserList(ctx context.Context) (*output.ListUsers, error) {
```

Comment:
[must] interfaceと実装が異なってるので修正必須です！

```suggestion
func (u *userUsecase) List(ctx context.Context) (*output.ListUsers, error) {
```